src/Bundle.ts

```diff
@@ -104,54 +385,115 @@ export default class Bundle {
 		).then(addons => addons.filter(Boolean).join(sep));
 	}
 
-	getPathRelativeToEntryDirname (resolvedId: string): string {
-		if (isRelative(resolvedId) || isAbsolute(resolvedId)) {
-			const entryDirname = dirname(this.entryModule.id);
-			const relativeToEntry = normalize(relative(entryDirname, resolvedId));
+	private setIdentifierRenderResolutions (options: OutputOptions) {
+		const used = blank();
+
+		let dynamicImportMechanism: DynamicImportMechanism;
+		const es = options.format === 'es'
 
-			return isRelative(relativeToEntry)
-				? relativeToEntry
-				: `./${relativeToEntry}`;
+		if (!es) {
+			if (options.format === 'cjs') {
+				dynamicImportMechanism = {
+					left: 'Promise.resolve(require(',
+					right: '))'
+				};
+			} else if (options.format === 'amd') {
+				dynamicImportMechanism = {
+					left: 'new Promise(function (resolve, reject) { require([',
+					right: '], resolve, reject) })'
+				}
+			}
 		}
 
-		return resolvedId;
-	}
+		if (this.graph.dynamicImport) {
+			this.orderedModules.forEach(module => {
+				module.dynamicImportResolutions.forEach((replacement, index) => {
```

Comment:
Maybe it is possible to extract this into a separate method?

Reply:
Sure